Dockerfile

```diff
@@ -58,11 +58,6 @@ RUN bash /tmp/env-config.sh
 #
 # This layer is populated with up-to-date files from
 # Calypso development.
-#
-# We remove apps, tests and desktop because they are not needed to
-# build or run calypso, but yarn will still install their
-# dependencies which end up bloating the image.
-# /apps/notifications is not removed because it is required by Calypso
 COPY . /calypso/
 RUN yarn install --immutable --check-cache
```

Comment:
Would it also make sense to separate the yarn install layer? Deps change rarely so we don't need to invalidate it by the `COPY` above.

```dockerfile
COPY package.json yarn.lock
RUN yarn install --immutable --check-cache
COPY . /calypso/
```

Reply:
Interesting idea! Do you think this will speed up builds even with `--pull`? I have to say I don't understand layers that well yet.

Reply:
> Do you think this will speed up builds even with --pull?

Not sure what a "`--pull`" build is, but in general, the `yarn` layer should depend only on the content of `package.json` and `yarn.lock` files – it stays cached as long as those stay unchanged.

> I have to say I don't understand layers that well yet.

I've only scratched the surface of the layer caching, but this idea actually came from the official docs I read the other day: https://docs.docker.com/build/cache/#order-your-layers!

Reply:
Thanks for the link! I think `--pull` makes Docker "pull" images from the remote which haven't been fetched yet. As a result, I think the docker image is built from scratch for each commit